smartsnippets/admin.py

```diff
@@ -79,6 +81,31 @@ def clean_template_path(self):
             raise ValidationError(e)
         return path
 
+    def clean(self):
+        clean_result = super(SnippetForm, self).clean()
+        self.validate_unique_variable_names()
+        return clean_result
+
+    def validate_unique_variable_names(self):
+        """ Validates name uniqueness over all variable inlines. """
+        unique_variable_names = set()
+        duplicate_variable_names = set()
```

Comment:
(Optional) Fancier way of doing the same thing:

  duplicate_variable_names = [
      var_name
      for var_name, count in collections.Counter(['foo', 'bar', 'foo']).iteritems() 
      if count > 1
  ]